static/js/main.js

```diff
@@ -49,6 +49,133 @@
+  var loadingSpinner = contributorCard.querySelector('.spinner')
+
+  if (window.IntersectionObserver) {
+    var observer = new window.IntersectionObserver(function (entries) {
+      entries.forEach(function (entry) {
+        if (entry.intersectionRatio > 0.5) {
+          // In viewport, fetch a random contributor
+          fetchRandomContributor()
+
+          observer.unobserve(entry.target)
+        }
+      })
+    },
+    { threshold: 0.5 }
+    )
+
+    observer.observe(document.querySelector('footer'))
+  } else {
+    // Does not support IntersectionObserver
+    fetchRandomContributor()
+  }
+
+  function fetchRandomContributor () {
+    var maxContributors
+    var fetchDate
+    var needToRefetch = false
+
+    if (window.localStorage) {
+      maxContributors = window.localStorage.getItem('max_contributors')
+      fetchDate = parseInt(window.localStorage.getItem('fetch_date'), 10)
+    }
+
+    // If fetch date is a month old (2592000000 ms === 30 days)
+    if (Date.now() - fetchDate >= 2592000000) {
+      needToRefetch = true
+    }
+
+    // If localStorage and data is less than 1 month old, fetch 1 time
+    if (maxContributors && !needToRefetch) {
+      getContributor(Math.floor(Math.random() * Math.floor(parseInt(maxContributors))) + 1)
+    } else {
+      getMaxContributors(function (randomPage, lastPage) {
+        getContributor(randomPage)
+
+        if (window.localStorage) {
+          window.localStorage.setItem('max_contributors', lastPage)
+        }
+      })
+    }
+  }
+
+  function getMaxContributors (callback) {
+    var xhr = new window.XMLHttpRequest()
+    xhr.responseType = 'json'
+
+    xhr.open('GET', 'https://api.github.com/repos/nodejs/node/contributors?per_page=1')
+    xhr.send()
+    xhr.addEventListener('load', function () {
+      if (xhr.status !== 200) {
+        return contributorCard.remove()
+      }
+
+      // Get Headers Links last page to generate a random contributor
+      var links = linkParser(xhr.getResponseHeader('Link'))
+      var randomPage = Math.floor(Math.random() * Math.floor(parseInt(links.last.page, 10))) + 1
+
+      if (window.localStorage) {
+        window.localStorage.setItem('fetch_date', Date.now())
+      }
+
+      callback(randomPage, links.last.page)
+    })
+  }
+
+  function getContributor (randomPage) {
+    var xhr = new window.XMLHttpRequest()
+    xhr.responseType = 'json'
+
+    xhr.open('GET', 'https://api.github.com/repos/nodejs/node/contributors?per_page=1&page=' + randomPage)
+    xhr.send()
+    xhr.addEventListener('load', function () {
+      if (xhr.status !== 200) {
+        return contributorCard.remove()
+      }
+
+      var contributor = xhr.response[0]
+
+      // Remove loading spinner and show avatar
+      loadingSpinner.remove()
+      contributorAvatar.classList.remove('hidden')
+      // Set new values
+      contributorAvatar.src = contributor.avatar_url
```

Comment:
We should append `&s=80` since we know we want this to be 40x40, thus it will still look good on retina